### PR TITLE
fix: Update instance size options for API Management service - `avm/res/api-management/service`

### LINKED ISSUE #1628: [AVM Module Issue]: API Management consumption plan requires skuCount 0
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/api-management/service

### (Optional) Module Version

_No response_

### Description

If you want to deploy a consumption sku for Api Management it fails because the skuCount is required to be set to 0. Error is 
_For Consumption SKU Type capacity must be specified as 0._

At the moment you can't specify skuCount as zero as the code for that is:

```
@description('Optional. The instance size of this API Management service.')
@allowed([
  1
  2
])
param skuCount int = 1
```

Please add allowed 0 to this list.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by cvs79**
Added '0' as an allowed value for instance size selection. Fixes #1628 

## Pipeline Reference

| Pipeline |
| -------- |
|          |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [X] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/api-management/service/main.bicep [hunk 1]
```diff
@@ -66,6 +66,7 @@ param sku string = 'Developer'
 
 @description('Optional. The instance size of this API Management service.')
 @allowed([
+  0
```
Comment (AlexanderSehr) [inline avm/res/api-management/service/main.bicep:69]: Beware, this also requires a run of the `Set-AVMModule` script as this changes the template hash & the allowed set in the readme (:

Comment (tony-box) [reply avm/res/api-management/service/main.bicep:69]: I am going to run it in my forked repo so I can use the Set-AVMModule script and re-do the PR. Thanks for the heads up!

Comment (AlexanderSehr) [reply avm/res/api-management/service/main.bicep:69]: Hey @tony-box, I'm confused. Is @cvs79 your stage name, or how come this user wouldn't just run set script? 😄

Comment (cvs79) [reply avm/res/api-management/service/main.bicep:69]: I thought I did but apparently it was not in my pr.

Comment (AlexanderSehr) [reply avm/res/api-management/service/main.bicep:69]: All good. The readme is now correctly updated, but for some reason, the `bicep build` was skipped. The template hash should update from `4652975451954294914` to `7048259745072727177`

Comment (cvs79) [reply avm/res/api-management/service/main.bicep:69]: Done. Probably some tooling issue on my end

**Comment by cvs79** [comment]
@tony-box  Can we get this merged?

**Comment by tony-box** [comment]
looks good. merging. sorry for the delay

**Review by tony-box — APPROVED** [review]

**Pull request merged**